web/packages/teleterm/src/ui/services/keyboardShortcuts/keyboardShortcutsService.ts

```diff
@@ -136,10 +159,13 @@ export class KeyboardShortcutsService {
   private mapAcceleratorsToActions(): void {
     this.acceleratorsToActions.clear();
     Object.entries(this.shortcutsConfig).forEach(([action, accelerator]) => {
-      this.acceleratorsToActions.set(
-        accelerator,
-        action as KeyboardShortcutAction
-      );
+      if (!accelerator) {
+        return;
+      }
```

Comment:
I think we should add a comment explaining that it's possible for an action to have no accelerator if the same accelerator is used for two different actions.

But in general a better choice (if feasible) would be to design the service in such a way that it returns early with an error rather than always having to remember about a scenario where an invalid value was provided.

In this case this could mean using a default accelerator for affected actions and showing a warning or an error.

Reply:
>I think we should add a comment explaining that it's possible for an action to have no accelerator if the same accelerator is used for two different actions.

In this particular case, empty accelerator means that someone provided an empty string in the config file, so the shortcut is disabled.

>But in general a better choice (if feasible) would be to design the service in such a way that it returns early with an error rather than always having to remember about a scenario where an invalid value was provided.

Good idea. I think we can remove `mapAcceleratorsToActions` from the constructor and let the initialization code call it (then I'd rename to sth like `initializeAccelerators`). This function could be called multiple times, for example when config was changed. This funtion would return an array of duplicates to show them in the UI.
WDYT?

> In this case this could mean using a default accelerator for affected actions and showing a warning or an error.

I'd skip the default accelerator part, if the user wanted to change the accelerator it doesn't make much sense to revert to the default IMO.

Reply:
> In this particular case, empty accelerator means that someone provided an empty string in the config file, so the shortcut is disabled.

It looks like I totally misread the code then, it doesn't represent a defensive check due to an invalid value being propagated through the program but a totally valid choice by the user. I think we can just document it and move on without refactoring the service. Given that my wrong assumptions were corrected, I don't understand what pain point the refactoring you proposed would help with here. 😅

Reply:
I think the pain point is that we keep an array of accelerators for each action (`Map<string, KeyboardShortcutAction[]>()` only to show the duplicates and in `getShortcutAction` we always use the first one.
But I think it is not a huge deal and we can leave it as it is.